lib/spack/spack/package.py

```diff
@@ -1494,26 +1543,46 @@ def build_process():
                     self.log()
 
                 # Run post install hooks before build stage is removed.
+                # (generates modules)
                 spack.hooks.post_install(self.spec)
 
             # Stop timer.
             self._total_time = time.time() - start_time
             build_time = self._total_time - self._fetch_time
 
-            tty.msg("Successfully installed %s" % self.name,
-                    "Fetch: %s.  Build: %s.  Total: %s." %
-                    (_hms(self._fetch_time), _hms(build_time),
-                     _hms(self._total_time)))
+            if self.name in setup:
+                tty.msg("Successfully setup %s" % self.name,
+                        "Config file is %s" % spconfig_fname)
+            else:
+                tty.msg("Successfully installed %s" % self.name,
+                        "Fetch: %s.  Build: %s.  Total: %s." %
+                        (_hms(self._fetch_time), _hms(build_time),
+                         _hms(self._total_time)))
+    
             print_pkg(self.prefix)
 
             # preserve verbosity across runs
             return echo
+        # ------------------------- End of build_process()
+
+        if is_setup:
+            keep_prefix = True
 
         # hook that allow tests to inspect this Package before installation
         # see unit_test_check() docs.
         if not self.unit_test_check():
             return
 
+        try:
```

Comment:
@citibeth This needs to be merged with the logic in the next `try` block; I plan on doing that myself, but I want to know what issue you are trying to avoid with this.

Reply:
Yes I agree they need to be merged; looks like a few lines of code got duplicated somehow.

The issue here is one a of a number of subtle, small but annoying things that crop up with Spack Install.  Remember how `spack setup` is usually used:
```
spack install --setup mypackage mypackage
cd mypackage
mkdir build
../../mypackage-setup.py ..
make install
```
At this point, the `make install` command installs into the install directory created in line 1577 in the piece of code we're looking at.  Now suppose the user decides that `mypackage` needs another dependency.  So she edits `mypackage/package.py`, then does:
```
spack install --setup mypackage mypackage
cd mypackage/build
../../mypackage-setup.py ..
make install
```
This is a natural part of the iterative process.  But it will cause an annoying error if you don't have the check for `is_setup` in line 1582.  You shouldn't have to delete / uninstall your `setup` package just because you re-ran Spack.  (Or put another way: Spack was making the assumption that `spack install` writes into an install directory.  That assumption is no longer valid with `spack setup`, and management of the install directory is punted to the developer, i.e. the user of Spack).

Does this help?